### ISSUE TITLE
use go generate

### ISSUE THREAD

**Issue by langston-barrett** [acmacalister/skittles#6]
By its very nature, a lot of the code in this library is redundant. We could generate it [using go generate](https://golang.org/cmd/go/#hdr-Generate_Go_files_by_processing_source). 

The big advantage would be (after the refactor in #4), we could generate _all_ combinations of effects like bold, blink, underline, inverse with _all_ different colors.

**Comment by acmacalister** [comment]
agreed. It has been some time since I have played with go generate. If you already have something going, awesome. If not, I will try and look at this weekend.
